program/src/processor/deactivate_stake.rs

```diff
@@ -1,21 +1,136 @@
-use solana_program::{entrypoint::ProgramResult, pubkey::Pubkey};
+use std::num::NonZeroU64;
 
-use crate::instruction::accounts::{Context, DeactivateStakeAccounts};
+use solana_program::{
+    clock::Clock, entrypoint::ProgramResult, program_error::ProgramError, pubkey::Pubkey,
+    sysvar::Sysvar,
+};
+
+use crate::{
+    error::StakeError,
+    instruction::accounts::{Context, DeactivateStakeAccounts},
+    require,
+    state::{find_stake_pda, Config, Stake},
+};
+
+/// Helper to calculate the maximum amount that can be deactivated.
+#[inline(always)]
+fn get_max_deactivation_amount(total_amount: u64, basis_points: u16) -> Result<u64, ProgramError> {
+    total_amount
+        .checked_mul(basis_points as u64)
+        .ok_or(ProgramError::ArithmeticOverflow)?
+        .checked_div(10_000)
+        .ok_or(ProgramError::ArithmeticOverflow)
+}
 
 /// Deactivate staked tokens for the validator.
 ///
 /// Only one deactivation may be in-flight at once, so if this is called
 /// with an active deactivation, it will succeed, but reset the amount and
 /// timestamp.
 ///
-/// 0. `[w]` Validator stake account
-/// 1. `[s]` Authority on validator stake account
+/// 0. `[]` Stake config account
+/// 1. `[w]` Validator stake account
+/// 2. `[s]` Authority on validator stake account
 ///
 /// Instruction data: amount of tokens to deactivate, as a little-endian u64
 pub fn process_deactivate_stake(
-    _program_id: &Pubkey,
-    _ctx: Context<DeactivateStakeAccounts>,
-    _amount: u64,
+    program_id: &Pubkey,
+    ctx: Context<DeactivateStakeAccounts>,
+    amount: u64,
 ) -> ProgramResult {
+    // Account valuidation.
```

Comment:
Sorry! Spotted this on both PR's. 😁

```suggestion
    // Account validation.
```